Source/BugsnagMetadata.m

```diff
@@ -94,33 +94,107 @@ - (NSDictionary *)toDictionary {
     }
 }
 
+/**
+ * Add a single key/value to a metadata Tab/Section.
+ */
 - (void)addAttribute:(NSString *)attributeName
            withValue:(id)value
-       toTabWithName:(NSString *)sectionName
-{
+       toTabWithName:(NSString *)tabName {
+    
+    bool metadataChanged = false;
     @synchronized(self) {
-        if (value) {
+        if (value && value != [NSNull null]) {
             id cleanedValue = BSGSanitizeObject(value);
             if (cleanedValue) {
-                NSDictionary *section = [self getMetadata:sectionName];
+                // Value is OK, try and set it
+                NSMutableDictionary *section = [self getMetadata:tabName];
                 if (!section) {
-                    section = [NSMutableDictionary new];
-                    [[self dictionary] setObject:section forKey:sectionName];
+                    section = [@{} mutableCopy];
+                    [[self dictionary] setObject:section forKey:tabName];
```

Comment:
Not necessary bad, but why the change from making a dictionary using the constructor to using a copy?

Reply:
I'm not actually sure why I made the change.  I agree, though, why jump through an unnecessary hoop?  Changed back.